src/Pages/PokemonView/components/HeaderPokemonView/HeaderPokemonView.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useEffect, useState } from "react";
+import { SortBy, sortByOptions, ViewType } from "./consts";
+import SearchBar from "../../../../Components/SearchBar/SearchBar";
+import DropDown from "../../../../Components/DropDown/DropDown";
+import {
+  DropDownContainer,
+  getDropdownStyle,
+  HeaderContainer,
+  TabAndSearchBarContainer,
+} from "./styles";
+import Tab from "./Tab/Tab";
+import { HeaderPokemonViewProps } from "./types";
+import { sortType } from "../../../../Components/Table/consts";
+
+const HeaderPokemonView = ({
+  setSearchBy,
+  setSortOrder,
+  viewType,
+  setViewType,
+}: HeaderPokemonViewProps) => {
+  const [sortBy, setSortBy] = useState("");
+  useEffect(() => {
+    switch (sortBy) {
+      case SortBy.NAME_AZ:
+        setSortOrder({ col: "name", order: sortType.ASC });
+        break;
+      case SortBy.NAME_ZA:
+        setSortOrder({ col: "name", order: sortType.DESC });
+        break;
+      case SortBy.POWER_HL:
+        setSortOrder({ col: "Power", order: sortType.DESC });
+        break;
+      case SortBy.POWER_LH:
+        setSortOrder({ col: "Power", order: sortType.ASC });
+        break;
+      case SortBy.HP_HL:
+        console.log("DESC");
+        setSortOrder({ col: "hp", order: sortType.DESC });
+        break;
+      case SortBy.HP_LH:
+        console.log("ASC");
+        setSortOrder({ col: "hp", order: sortType.ASC });
+        break;
+      default:
+        setSortOrder({ col: "ID", order: sortType.ASC });
+    }
```

Comment:
add this:
```
export type SortOptionType {
   title: string;
   value: sortType
}

const [sortBy, setSortBy] = useState<SortOptionType>({})
```
and then you wont need the useEffect at all, when the user will chnage the sortType the state value will get the exact type that you need.
IF you dont understand what i meant ask me.